### ISSUE TITLE
Cannot `plan -target` to delete resources

### ISSUE THREAD

**Issue by jszwedko** [hashicorp/terraform#3852]
Running `terraform plan -target` to target a delete of a resource results in terraform thinking that there is no diff.

Example config:
```
provider "aws" {
  region = "us-east-1"
}

resource "aws_instance" "web" {
  ami = "ami-60b6c60a"
  instance_type = "t2.micro"
}
```

Run plan/apply as normal.

Comment out the lines so you end up with:

```
provider "aws" {
  region = "us-east-1"
}

/* resource "aws_instance" "web" { */
/*   ami = "ami-60b6c60a" */
/*   instance_type = "t2.micro" */
/* } */
```

Run `terraform plan`, observe that it wants to delete the instance, run `terraform plan -target aws_instance.web`, observe that it thinks there are no changes.

E.g.:
```
ip-10-254-32-102:/tmp/terraform-test% terraform plan
Refreshing Terraform state prior to plan...

aws_instance.web: Refreshing state... (ID: i-0512c5bb)

The Terraform execution plan has been generated and is shown below.
Resources are shown in alphabetical order for quick scanning. Green resources
will be created (or destroyed and then created if an existing resource
exists), yellow resources are being changed in-place, and red resources
will be destroyed.

Note: You didn't specify an "-out" parameter to save this plan, so when
"apply" is called, Terraform can't guarantee this is what will execute.

- aws_instance.web


Plan: 0 to add, 0 to change, 1 to destroy.
ip-10-254-32-102:/tmp/terraform-test% terraform plan -target aws_instance.web
Refreshing Terraform state prior to plan...


No changes. Infrastructure is up-to-date. This means that Terraform
could not detect any differences between your configuration and
the real physical resources that exist. As a result, Terraform
doesn't need to do anything.
```

**Comment by jen20** [comment]
@jszwedko Thanks for reporting this. I can confirm I can reproduce your error (using an `aws_iam_user` resource instead of an `aws_instance`), and the same reproduction steps. I think I see what is wrong here - I will investigate further tomorrow.

**Comment by jszwedko** [comment]
Thanks @jen20 !

**Comment by jen20** [comment]
After applying the fix in #3912, my previous test gives the (expected) output:

With resource:
```
$ terraform apply
aws_iam_user.lb: Creating...
  arn:       "" => "<computed>"
  name:      "" => "loadbalancer"
  path:      "" => "/system/"
  unique_id: "" => "<computed>"
aws_iam_user.lb: Creation complete

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

The state of your infrastructure has been saved to the path
below. This state is required to modify and destroy your
infrastructure, so keep it safe. To inspect the complete state
use the `terraform show` command.

State path: terraform.tfstate
```

After commenting out the resource:

```
$ terraform plan -target aws_iam_user.lb
Refreshing Terraform state prior to plan...

aws_iam_user.lb: Refreshing state... (ID: loadbalancer)

The Terraform execution plan has been generated and is shown below.
Resources are shown in alphabetical order for quick scanning. Green resources
will be created (or destroyed and then created if an existing resource
exists), yellow resources are being changed in-place, and red resources
will be destroyed.

Note: You didn't specify an "-out" parameter to save this plan, so when
"apply" is called, Terraform can't guarantee this is what will execute.

- aws_iam_user.lb


Plan: 0 to add, 0 to change, 1 to destroy.
```

**Comment by jszwedko** [comment]
:clap: thank you @jen20 !
